packages/bot/src/transformers/interaction.ts

```diff
@@ -220,55 +223,62 @@ const baseInteraction: Partial<Interaction> & BaseInteraction = {
   },
 }
 
-export function transformInteraction(bot: Bot, payload: DiscordInteraction): Interaction {
-  const guildId = payload.guild_id ? bot.transformers.snowflake(payload.guild_id) : undefined
-  const user = bot.transformers.user(bot, payload.member?.user ?? payload.user!)
+export function transformInteraction(bot: Bot, payload: { interaction: DiscordInteraction; shardId: number }): Interaction {
+  const guildId = payload.interaction.guild_id ? bot.transformers.snowflake(payload.interaction.guild_id) : undefined
+  const user = bot.transformers.user(bot, payload.interaction.member?.user ?? payload.interaction.user!)
 
   const interaction: Interaction = Object.create(baseInteraction)
   const props = bot.transformers.desiredProperties.interaction
   interaction.bot = bot
   interaction.acknowledged = false
 
-  if (props.id && payload.id) interaction.id = bot.transformers.snowflake(payload.id)
-  if (props.applicationId && payload.application_id) interaction.applicationId = bot.transformers.snowflake(payload.application_id)
-  if (props.type && payload.type) interaction.type = payload.type
-  if (props.token && payload.token) interaction.token = payload.token
-  if (props.version && payload.version) interaction.version = payload.version
-  if (props.locale && payload.locale) interaction.locale = payload.locale
-  if (props.guildLocale && payload.guild_locale) interaction.guildLocale = payload.guild_locale
+  if (props.id && payload.interaction.id) interaction.id = bot.transformers.snowflake(payload.interaction.id)
+  if (props.applicationId && payload.interaction.application_id)
+    interaction.applicationId = bot.transformers.snowflake(payload.interaction.application_id)
+  if (props.type && payload.interaction.type) interaction.type = payload.interaction.type
+  if (props.token && payload.interaction.token) interaction.token = payload.interaction.token
+  if (props.version && payload.interaction.version) interaction.version = payload.interaction.version
+  if (props.locale && payload.interaction.locale) interaction.locale = payload.interaction.locale
+  if (props.guildLocale && payload.interaction.guild_locale) interaction.guildLocale = payload.interaction.guild_locale
+  if (props.guild && payload.interaction.guild)
+    // @ts-expect-error payload.interaction.guild is a Partial<DiscordGuild>
```

Comment:
Are we sure that it's okay to pass Partial stuff to the transformer?

Reply:
The guild transformers has payload checks in the file, so there shouldn't be an issue with this